src/Gemini/Modules/Shell/Commands/SaveAllFilesCommandHandler.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Gemini.Framework;
+using Gemini.Framework.Commands;
+using Gemini.Framework.Services;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Gemini.Modules.Shell.Commands
+{
+    [CommandHandler]
+    public class SaveAllFilesCommandHandler : CommandHandlerBase<SaveAllFilesCommandDefinition>
+    {
+        private readonly IShell _shell;
+
+        [ImportingConstructor]
+        public SaveAllFilesCommandHandler(IShell shell)
+        {
+            _shell = shell;
+        }
+
+        public override async Task Run(Command command)
```

Comment:
One nitpick, remove the async and return TaskUtility.Completed here, otherwise the compiler will give warnings.